pkg/extensions/util.go

```diff
@@ -116,44 +120,40 @@ func diffExtensions(oldExts, newExts ecv1beta1.Extensions) DiffResult {
 		}
 	}
 
-	var added, removed, modified []ecv1beta1.Chart
+	var results []diffResult
 
-	// find removed and modified charts.
-	for name, oldChart := range oldCharts {
-		newChart, exists := newCharts[name]
-		if !exists {
-			// chart was removed.
-			removed = append(removed, oldChart)
-		} else if !reflect.DeepEqual(oldChart, newChart) {
-			// chart was modified.
-			modified = append(modified, newChart)
+	for name, newChart := range newCharts {
+		r := diffResult{
+			Ext: newChart,
+		}
+		oldChart, ok := oldCharts[name]
+		if !ok {
+			// chart was added.
+			r.Action = actionInstall
+		} else {
+			r.Action = actionUpgrade
+			r.NeedsUpgrade = !reflect.DeepEqual(oldChart, newChart)
```

Comment:
any reason to not just `continue` here and skip this extension? seems weird to have the action be `upgrade` if the extensions doesn't need an upgrade

Reply:
because otherwise the condition will be missing from the CR

Reply:
is that a problem though? i think it's more confusing to had the CR show the extension was upgraded when it didn't need to be upgraded.

Reply:
I guess if we want to be verbose, we can at least have a 4th condition to indicate it was a no-op, or maybe just a log.

Reply:
nevermind, i guess this is fine

Reply:
That's how addons are processed. We indicate they were upgraded even if there was no diff. So this is consistent.